### DISCUSSION TITLE
KubeVPN client for windows ad AF_UNIX requirement

### DISCUSSION THREAD

**Discussion by vmaglietta** [kubenetworks/kubevpn#331] [Ideas]
After some digging I found that KubeVPN windows client can run only if the Win OS version is compliant with AF_UNIX feature otherwise the daemon cannot run. Runing a simple command `kubevpn status` hangs forever with no output while `kubevpn daemon` exits a more clear error. It should be nice if such condition is detected by the program exiting with a clear message about the OS incompatibility.

Morevoer is there any plan to make the windows client compliant with older OS without AF_UNIX support such as Windows server 2012 or 2016? I know this it not a simple task.

Thank you so much.

**Comment by wencaiwulue** [comment]
@vmaglietta i found two related issue about [grpc](https://github.com/grpc/grpc/issues/22285) and [go](https://github.com/golang/go/issues/57004), go1.21 not support windows server 2012, and project using 1.22.1

**Reply by vmaglietta** [reply]
@wencaiwulue many thanks for the info! Is there any older KubeVPN version that was developed using go 1.17, ... just for fast testing on my old OSes stuff ;-)

**Reply by vmaglietta** [reply]
Well thanks to your hints I found that **v1.2.0** is the latest compatible with Windows server 2016, I will check 2012 in the next days and keep you updated. @wencaiwulue  you're a superhero!
